iFixit/react-commerce
# Facets: Hide refinement items whose count equals hits

If a refinement item (aka a facet option) is applicable to the same
number of products as the number of products in the current view, then
that refinement item is essentially a no-op and we should hide it.

One exception: Refinement items that are currently selected should still
be displayed.

## QA
Click through some different product list views (tools and parts, mobile and desktop). Make sure that no facet options are ever displayed if they apply to the same number of products as are displayed in the current view, and no other facet options in that facet are selected. (This differs slightly from the original issue; see https://github.com/iFixit/react-commerce/pull/399#issuecomment-1179234914 and the conversation leading up to it).

CC @sterlinghirsh 

Closes #388
